docs/debug-mode.rst

```diff
@@ -0,0 +1,17 @@
+Debug Mode
+==========
+
+HPy includes a debug mode that may be activated at *run time* with
+*no overhead*. This is possible because the whole of the HPy API is provided
```

Comment:
```suggestion
HPy includes a debug mode which includes a lot of useful run-time checks to
ensure that C extensions use the API correctly. The major points of the debug mode are:

    1. no special compilation flags are required: it is enough to compile the extension 
       with the Universal ABI.
    
    2. The debug mode can be activated at *import time*, and it can be activated
       per-extension.
    
    3. You pay the overhead of the debug mode only if you use it. Extensions loaded 
       without the debug mode run at full speed.

This is possible because the whole of the HPy API is provided
```